Salesforce/Inbound/Compiler/EntityCompiler.php

```diff
@@ -88,19 +90,27 @@ public function compile(SObject $object, string $connectionName = 'default', $va
         $metas    = $connection->getMetadataRegistry()->findMetadataBySObject($object);
 
         foreach ($metas as $metadata) {
-            $class   = $metadata->getClassName();
+            $class = $metadata->getClassName();
+            $this->logger->info('EntityCompiler->compile()-010 $class = '.$class);
             $manager = $this->registry->getManagerForClass($class);
             /** @var ClassMetadata $classMetadata */
             $classMetadata = $manager->getClassMetadata($class);
             $entity        = $this->convertToEntity(
                 $object,
-                $metadata
+                $metadata,
+                $deliveryMethod
             );
 
-            // Check if the entity is meant for this connection, if the connection value for the entity is null,
-            // don't check, allow the entity to be created, given that validation passes
+            // Check if the entity is not meant for this connection, allow the entity to be created, given that validation passes
             $connectionProp = $metadata->getConnectionNameField();
-
+            if (null !== $connectionProp) {
+                $this->logger->info('EntityCompiler->compile()-021 - Not Null');
```

Comment:
need to be more specific than '-021 - Not null'.  Maybe 'connectionProp is null'